packages/protocol/contracts/L1/libs/LibProvingAlt.sol

```diff
@@ -0,0 +1,365 @@
+                // such changes would require additional if-else logic.
+                ts.prover = blk.assignedProver;
+            } else {
+                // In scenarios where this transition is not the first one, we
+                // straightforwardly reset the transition prover to address
+                // zero.
+                ts.prover = address(0);
+
+                // Furthermore, we index the transition for future retrieval.
+                // It's worth emphasizing that this mapping for indexing is not
+                // reusable. However, given that the majority of blocks will
+                // only possess one transition — the correct one — we don't need
+                // to be concerned about the cost in this case.
+                state.transitionIds[blk.blockId][tran.parentHash] = tid;
+            }
+        } else {
+            // A transition with the provided parentHash has been located.
+            ts = state.transitions[slot][tid];
+            if (ts.tier < meta.minTier) {
+                revert L1_UNEXPECTED_TRANSITION_TIER();
+            }
+        }
+
+        // The new proof must meet or exceed the minimum tier required by the
+        // block or the previous proof; it cannot be on a lower tier.
+        if (proof.tier == 0 || proof.tier < meta.minTier || proof.tier < ts.tier) {
+            revert L1_INVALID_TIER();
+        }
+
+        // Retrieve the tier configurations. If the tier is not supported, the
+        // subsequent action will result in a revert.
+        ITierProvider.Tier memory tier =
+            ITierProvider(resolver.resolve("tier_provider", false)).getTier(proof.tier);
+
+        _checkProverPermission(blk, ts, tid, tier);
+
+        // We must verify the proof, and any failure in proof verification will
+        // result in a revert.
+        //
+        // It's crucial to emphasize that the proof can be assessed in two
+        // potential modes: "proving mode" and "contesting mode." However, the
+        // precise verification logic is defined within each tier'IVerifier
+        // contract implementation. We simply specify to the verifier contract
+        // which mode it should utilize - if the new tier is higher than the
+        // previous tier, we employ the proving mode; otherwise, we employ the
+        // contesting mode (the new tier cannot be lower than the previous tier,
+        // this has been checked above).
+        //
+        // It's obvious that proof verification is entirely decoupled from
+        // Taiko's core protocol.
+        {
+            address verifier = resolver.resolve(tier.verifierName, true);
+            // The verifier can be address-zero, signifying that there are no
+            // proof checks for the tier. In practice, this only applies to
+            // optimistic proofs.
+            if (verifier != address(0)) {
+                bool isContesting = proof.tier == ts.tier && tier.contestBond != 0;
```

Comment:
Tiny thing, but GuardianVerifier does not check `isContesting` inside - as it just checks who is the prover (as indeed the "guardian_prover"). And in case it will in the future (i guess not) it is better to remove this check.

```suggestion
                bool isContesting = proof.tier == ts.tier;
```

Reply:
This change should be revert. Done.